### PR TITLE
Autoproxy

### PR DESCRIPTION
**Pull request by ypwalter**
@ShakoHo, r?

### CODE REVIEW

File: lib/profiler/mitmDumpProfiler.py [hunk 1]
```diff
@@ -36,8 +38,26 @@ def start_recording(self):
                 replay_cmd_list = ["mitmdump", "--no-http2", "-S", flow_file_fp, "--no-pop", "--norefresh"]
             self.process = subprocess.Popen(replay_cmd_list)
 
+        if sys.platform == "linux2":
+            subprocess.call(["gsettings", "set", "org.gnome.system.proxy",       "mode", "manual"])
+            subprocess.call(["gsettings", "set", "org.gnome.system.proxy.http",  "host", "127.0.0.1"])
+            subprocess.call(["gsettings", "set", "org.gnome.system.proxy.http",  "port", "8080"])
```
Comment (askeing) [inline lib/profiler/mitmDumpProfiler.py]: multiple space

**Comment by ypwalter** [comment]
@askeing r?

File: lib/profiler/mitmDumpProfiler.py [hunk 2]
```diff
@@ -36,8 +38,26 @@ def start_recording(self):
                 replay_cmd_list = ["mitmdump", "--no-http2", "-S", flow_file_fp, "--no-pop", "--norefresh"]
             self.process = subprocess.Popen(replay_cmd_list)
 
+        if sys.platform == "linux2":
+            subprocess.call(["gsettings", "set", "org.gnome.system.proxy", "mode", "manual"])
+            subprocess.call(["gsettings", "set", "org.gnome.system.proxy.http", "host", "127.0.0.1"])
+            subprocess.call(["gsettings", "set", "org.gnome.system.proxy.http", "port", "8080"])
+            subprocess.call(["gsettings", "set", "org.gnome.system.proxy.https", "host", "127.0.0.1"])
+            subprocess.call(["gsettings", "set", "org.gnome.system.proxy.https", "port", "8080"])
+        else:
```
Comment (askeing) [inline lib/profiler/mitmDumpProfiler.py]: i'll suggest using `elif 'win32'` and `elif 'darwin'` to check the os, and `else` for handling the other cases.

File: lib/profiler/mitmDumpProfiler.py [hunk 3]
```diff
@@ -36,8 +38,26 @@ def start_recording(self):
...
+            proxy_setting_cmd = ["reg", "add", "\"HKCU\\Software\\Microsoft\\Windows\\CurrentVersion\\Internet Settings\"", "/v", "ProxyServer", "/t", "REG_SZ", "/d", "127.0.0.1:8080", "/f"]
+            os.system(" ".join(proxy_enable_cmd))
+            os.system(" ".join(proxy_setting_cmd))
+            os.system("cmd /c start \"\" /min \"C:\\Program Files\\Internet Explorer\\iexplore.exe\"")
+            time.sleep(5)
+            os.system("taskkill /T /IM iexplore.exe /F")
+
     def stop_recording(self, **kwargs):
         if sys.platform == "win32":
+            proxy_disable_cmd = ["reg", "add", "\"HKCU\\Software\\Microsoft\\Windows\\CurrentVersion\\Internet Settings\"", "/v", "ProxyEnable", "/t", "REG_DWORD", "/d", "0", "/f"]
```
Comment (askeing) [inline lib/profiler/mitmDumpProfiler.py]: the `sys.platform` checking should be the same as `start_recording()` method

File: lib/profiler/mitmDumpProfiler.py (hunk 2 above)
Comment (askeing) [inline lib/profiler/mitmDumpProfiler.py]: If we don't have the way to start recording on `darwin`, we can raise `NotImplementedError`
Ref: https://docs.python.org/2/library/exceptions.html#exceptions.NotImplementedError

**Comment by askeing** [comment]
LGTM

**Pull request merged**